### PR TITLE
kucoin: update margin trading / lending to v3

### PR DESCRIPTION
**Pull request by sc0Vu**
fix: ccxt/ccxt#18510
In this PR, I update both margin trading and lending to v3.

### CODE REVIEW

File: ts/src/kucoin.ts [hunk 1]
```diff
@@ -3965,40 +3998,27 @@ export default class kucoin extends Exchange {
...
-        let method = undefined;
         const timeInForce = this.safeStringN (params, [ 'timeInForce', 'type', 'borrowStrategy' ], 'IOC');
-        let timeInForceRequest = undefined;
-        if (symbol === undefined) {
-            method = 'privatePostMarginBorrow';
-            timeInForceRequest = 'type';
-        } else {
+        if (isIsolated) {
+            if (symbol === undefined) {
+                throw new Error ('symbol is required when isolated margin');
```
Comment (carlosmiei) [inline ts/src/kucoin.ts:4004]: @sc0Vu we should not throw an error that generic, why not `BadRequest` for instance?

File: ts/src/kucoin.ts [hunk 2]
```diff
@@ -4011,80 +4031,60 @@ export default class kucoin extends Exchange {
...
-        let method = undefined;
-        const sequence = this.safeString2 (params, 'sequence', 'seqStrategy', 'RECENTLY_EXPIRE_FIRST');
-        let sequenceRequest = undefined;
-        if (symbol === undefined) {
-            method = 'privatePostMarginRepayAll';
-            sequenceRequest = 'sequence';
-        } else {
+        if (isIsolated) {
+            if (symbol === undefined) {
+                throw new Error ('symbol is required when isolated margin');
```
Comment (carlosmiei) [inline ts/src/kucoin.ts:4054]: same here

File: ts/src/kucoin.ts (hunk 1 above)
Comment (sc0Vu) [reply ts/src/kucoin.ts:4004]: @carlosmiei how about ArgumentsRequired error?

Comment (carlosmiei) [reply ts/src/kucoin.ts]: @sc0Vu yeah it can be that as well

**Comment by carlosmiei** [comment]
LGTM ✅ 
```
 p kucoin borrowMargin USDT 1                
Python v3.10.9
CCXT v4.0.19
kucoin.borrowMargin(USDT,1)
{'amount': None,
 'currency': 'USDT',
 'datetime': '2023-07-15T13:16:04.197Z',
 'id': None,
 'info': {},
 'symbol': None,
 'timestamp': 1689426964197}
```
```
 p kucoin repayMargin USDT 1      
Python v3.10.9
CCXT v4.0.19
kucoin.repayMargin(USDT,1)
{'amount': 0.0,
 'currency': 'USDT',
 'datetime': '2023-07-15T13:17:50.500Z',
 'id': '64b29c7e4c89d70001e63456',
 'info': {'actualSize': '0',
          'orderNo': '64b29c7e4c89d70001e63456',
          'timestamp': 1689427070429},
 'symbol': None,
 'timestamp': 1689427070500}
```

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**